src/Subscription.ts

```diff
@@ -46,17 +48,15 @@ export class Subscription {
       const len = _subscriptions.length;
 
       while (++index < len) {
-        const sub = _subscriptions[index];
+        const sub: Subscription = _subscriptions[index];
         if (isObject(sub)) {
-          let trial = tryCatch(sub.unsubscribe).call(sub);
-          if (trial === errorObject) {
+          const [ok, e] = callPublicUnsubscribe(sub);
```

Comment:
A faster approach is to call a function that *only* has the try/catch block in it, that in turn calls another method that doesn't have any try/catch blocks in it (and optimizes better).


```js
class Foo() {
  start(x) {
    //do stuff here
    this.trySomeThing(x);
  }

  trySomeThing(x) {
    let result;
    try {
      result = this.someThing(x);
    } catch (err) {
      this.handleError(err);
    }
    this.handleSuccess(result);
  }

  handleSuccess(result) {
    // do more stuff here.
  }
}
```